### PR TITLE
Do not re-order index fields when sorting

### LINKED ISSUE #1195: order of fields for an index shouldn't be reordered in alphabetical order
### [REQUIRED] Environment info

All versions, all platforms.

**firebase-tools:**

6.6.0

**Platform:**

macOS

### [REQUIRED] Test case

With https://github.com/firebase/firebase-tools/pull/1168 this patch, fields of indexes is reordered in alphabetical order. 

we can't do this kind of operation properly now as index ordered is changed.

```
firebase firestore:indexes -P staging > firestore.indexes.json
firebase deploy --only firestore:indexes -P production
```

for example, when there is an index "last_name" ascending, "first_name". ascending, it'll be outputted as "first_name" ascending, "last_name" ascending. 

### [REQUIRED] Steps to reproduce

run `firebase firestore:indexes` 

### [REQUIRED] Expected behavior

fields of an index shouldn't be reordered in alphabetical order.

### [REQUIRED] Actual behavior

fields of an index is reordered in alphabetical order

### PR DESCRIPTION
**Pull request by samtstern**
### Description

Fixes #1195

### Scenarios Tested

See the changes made in tests.  This test fails with the code at `HEAD`. Also sanity checked with indexes on a live project.

### Sample Commands

N/A

### CODE REVIEW

File: src/firestore/indexes-sort.ts [hunk 1]
```diff
@@ -79,7 +79,11 @@ export function compareApiField(a: API.Field, b: API.Field): number {
     return aName.fieldPath.localeCompare(bName.fieldPath);
   }
 
-  return compareArrays(a.indexConfig.indexes || [], b.indexConfig.indexes || [], compareApiIndex);
```
Comment (samtstern) [inline src/firestore/indexes-sort.ts]: Previously `compareArrays` implicitly sorted all arguments first.  Now it does not, so we change two calls to do it explicitly.

**Comment by samtstern** [comment]
@bkendall can we do a `6.6.1` release once this is merged?  Oooh and I will actually remember to add a changelog entry this time.

**Comment by Memeriaj** [comment]
We were planning on doing a release today anyways.  Functions has something they want to fix, then we were going to do the release.

**Comment by samtstern** [comment]
Sorry I *did* do "Squash and Merge" but there is a known bug on GitHub where if you hit "Enter" instead of clicking the button with your mouse then it does a regular messy merge.  So I dirtied the git.

I hate that bug.

**Pull request merged**